### PR TITLE
V.1.6.0: Added multiples features

### PR DESCRIPTION
**Pull request by unrealmitch**
- Included option to avoid include issue ID in jira worklog description (see readme)
- Included option to split toggl description with some string (see readme)
- Included link / href in issue ID of popup
- Now, if search issue in description is enabled, first search in description and THEN search in Project Title (before just in description)
- Search also Issue ID in Project Name (Toggl) as in description (before just take the Project Name)
- Navigation buttons (back / next day in calendar)
- Button to refresh and reset / go today)
- Now it shows actual "in progress" task of toggl as "WIP" in popup (before showed as "too short")
- Different format (red bold) for task in "WIP" or "too short"
- Fixed TOTAL sum (before showed "too short" or "WIP", where there are some task with this duration)
- Now TOTAL says "No entries" when fetch finish (before said all time "Loading")
-Also show jira registered time in popup (at left of total times of task)
- Project ID now can contain numbers (Ex: PRO34-1)
- Also included option to enable/disable this previous feature, but disables in frontend
- Button to open Clockwork Free Jira Plugin (To show own worklogs between selected dates) and option to enable it
- Other mini improvements

It's my first conntact with Vue, so don't blame me pls ^^

Also I edited default option "Parse Jira issue from description" (default: false, now: true)

V.1.5.7
- Improved UI with @simoneldevig suggestions
- Fixed problem when Summary of Toggle Task is empty
- Now calendar navigation buttons move end and start date at same time (before just set interval to 1 day, and move the start)
- Reformatted source code (with prettier)
- Fixed Clockwork Free URL Plugin
- Added custom url jira plugin to allow user other plugins
- Solved bugs & other problems

V.1.5.8
- Fix problem when create worklog in jira with toggle task without description
- Restore NPM Packages from original repo (downgrade some packages)

V1.5.9
- Added option to keep dates (start and end) - Save and persistent dates

V1.6.0
- Added option to set Monday first day of the week (calendar picker)
- Reorganize options screen
- All default options set to false
- Remove the clockwork path by default

### CODE REVIEW

**Comment by simoneldevig** [comment]
@unrealmitch wow thanks for the huge PR!

I'll try to see if I can have a look at it in the weekend 😄

File: src/popup/App.vue [hunk 1]
```diff
@@ -233,22 +287,32 @@ export default {
...
       let _self = this;
       return new Promise(function (resolve, reject) {
         if (_self.jiraIssueInDescription) {
-          const parsedIssue = log.description.match(/^[A-Z]*-[0-9]*/);
+          const parsedIssue = _self.matchIssueId(log.description);
           if (parsedIssue) {
             resolve(parsedIssue[0]);
           }
-          reject(log);
+          //reject(log);  If don't find in description, search in project title
```
Comment (simoneldevig) [inline src/popup/App.vue:304]: **Suggestion**
Just remove the ```reject(log)```and keep the comment

**Review by simoneldevig — CHANGES_REQUESTED** [review]
Good job @unrealmitch!

Code's looking good and there was only a minor suggestion. 

I only have some comments to the functional improvements 😄 

- Different format (red bold) for task in "WIP" or "too short"
  - Can you change the color to the default red? (--md-theme-default-accent)
- Navigation buttons (back / next day in calendar)
  - Can you remove the background so the date navigation buttons and align them to the icon in the datepicker?
- Button to refresh and reset / go today)
  - Could you move the refresh button to the top bar and make the reset button to just a text that is positioned absolute in the top right?
- Button to open Clockwork Free Jira Plugin (To show own worklogs between selected dates) and option to enable it
  - Great idea! But maybe instead of tying it to a specific JIRA plugin then it could just be a link that could be edited in the settings page? Then it could be a link to the worklog by default and people would be able to insert a link to Clockwork, Tempo Planner etc. according to the plugin they use for time registration? 

There is a lot of link errors which Travis CI fails. You can autofix these by running ```npx eslint --ext .js,.vue src --fix``` in the console.

All in all I really like the improvements and only have stuff for the UI 🚀  I hacked an example together in the inspector to show how I think the UI should look 😄 

Screenshot 2021-06-07 at 07 56 38

**Comment by unrealmitch** [comment]
> 
> 
> Good job @unrealmitch!
> 
> Code's looking good and there was only a minor suggestion.
> 
> I only have some comments to the functional improvements 😄
> 
>     * Different format (red bold) for task in "WIP" or "too short"
>       
>       * Can you change the color to the default red? (--md-theme-default-accent)
> 
>     * Navigation buttons (back / next day in calendar)
>       
>       * Can you remove the background so the date navigation buttons and align them to the icon in the datepicker?
> 
>     * Button to refresh and reset / go today)
>       
>       * Could you move the refresh button to the top bar and make the reset button to just a text that is positioned absolute in the top right?
> 
>     * Button to open Clockwork Free Jira Plugin (To show own worklogs between selected dates) and option to enable it
>       
>       * Great idea! But maybe instead of tying it to a specific JIRA plugin then it could just be a link that could be edited in the settings page? Then it could be a link to the worklog by default and people would be able to insert a link to Clockwork, Tempo Planner etc. according to the plugin they use for time registration?
> 
> 
> There is a lot of link errors which Travis CI fails. You can autofix these by running `npx eslint --ext .js,.vue src --fix` in the console.
> 
> All in all I really like the improvements and only have stuff for the UI 🚀 I hacked an example together in the inspector to show how I think the UI should look 😄
> 
> Screenshot 2021-06-07 at 07 56 38

Wow, graeat suggestions, the UI looks amazing! I will try to implement all this weekend!

**Comment by unrealmitch** [comment]
I've had some free time to apply the UI changes that you described! Now it looks much better, thanks.

I've also fixed a couple of things, although Travis doesn't seem to like me, despite running said command on my PC with no problems :(

**Review by simoneldevig — CHANGES_REQUESTED** [review]
@unrealmitch just tested it out and looks pretty good, and nice feature with the date persistence 💯 

Could you maybe fix the following.

1. Fix the date and text alignment: 
Screenshot 2021-06-21 at 08 15 36
2. Make all the options false by default
Screenshot 2021-06-21 at 08 16 48
3. Remove the clockwork path by default and maybe remove the (Usually Clockwork Free) text from the checkbox so it's not tied to anything.
4. Move the Save Settings button to the bottom of the settings page instead of being in the middle 

Once the requested changes are fixed I will merge and release 👍

File: src/options/App.vue [hunk 2]
```diff
@@ -33,6 +38,21 @@
...
           </div>
+          <br>
+          <h3>Extra Options</h3>
+          <md-checkbox v-model="saveDates">Save dates (Persistent start and end dates)</md-checkbox><br>
+          <md-checkbox v-model="clockworkEnabled">Button to Jira Plugin (Usually Clockwork Free)</md-checkbox>
+          <md-field>
+            <label>URL Jira Plugin</label>
+            <md-input v-model="jiraPlugin" />
+          </md-field>
+          <!-- <br><label>Start weekday:</label>
```
Comment (simoneldevig) [inline src/options/App.vue:49]: @unrealmitch properly should remove the unused code

**Comment by unrealmitch** [comment]
Solved 2,3 and 4. But I didn't understand point 1 "Fix the date and text alignment". I keep de format of calendar of original plugin. Did you mean change alignment to center?

**Comment by unrealmitch** [comment]
I also added the option to start week on monday (Calendar Picker), as it was mentioned in  #21, and with the option to save dates #20, all issues would be closed (and with button of reset dates it will works well)

**Comment by simoneldevig** [comment]
@unrealmitch 1 is just the vertical alignment between label and value on the date input :-)

**Comment by unrealmitch** [comment]
In this way? Updated css in last commit.

image

**Comment by unrealmitch** [comment]
In this way? Updated css in last commit.

image

**Pull request merged**